magma/lib/magma/query/predicate/file_collection.rb

```diff
@@ -0,0 +1,53 @@
+require 'json'
+
+class Magma
+  class FileCollectionPredicate < Magma::ColumnPredicate
+    verb '::url' do
+      child String
+
+      extract do |table, identity|
+        table.first[column_name] ? table.first[column_name].map do |f|
+          Magma.instance.storage.download_url(
+            @model.project_name,
+            f["filename"]
+          )
+        end : nil
+      end
+    end
+
+    verb '::path' do
+      child String
+
+      extract do |table, identity|
+        table.first[column_name] ? table.first[column_name].map { |f| f["filename"] } : nil
+      end
+    end
+
+    verb '::original_filename' do
+      child String
+
+      extract do |table, identity|
+        table.first[column_name] ? table.first[column_name].map { |f| f["original_filename"] } : nil
+      end
+    end
+
+    verb '::all' do
```

Comment:
What is this verb for? Seems basically the same as '::url' above?

Reply:
nm, now i get it - i might make this `verb nil` (that is, match this if there are no following args) to avoid confusion with the model predicate's `::all`

Reply:
nm x2 i now see that i introduced this confusion in `file` already :cry:

Reply:
And it is different than `::url` because it includes all the data, like `original_filename`.